### DISCUSSION TITLE
Building open source and commercial projects

### DISCUSSION THREAD

**Discussion by jasenf** [abpframework/abp#16185] [General]
Hi all --

I think I already know the answer to this but I'd like some confirmation --

We'd like to build an open source application on top of the open source version of ABP, and then an additional commercial paid SaaS app using the commercial version of ABP that has additional features and functionality. 

Open source git projects on top of the free abp version are allowed correct?  

Including the pre-generated project code created from the abp.io /get-started page...

**Comment by hikalkan** [comment]
Hi @jasenf, 

Will you use ABP as NuGet packages in your application, or you will copy its source code and embed into your applicaton? I suppose you will use as packages. In that case, that is possible, yes. Commercial side is also possible.

**Reply by jasenf** [reply]
We would generate the original code with abp-suite, using only open-source (non-commercial) components via nuget packages.

Then build from there.

I'm pretty sure the licensing allows for this, just double checking..
